src/main/java/org/candlepin/audit/EventBuilder.java

```diff
@@ -83,14 +80,10 @@ public EventBuilder setEventData(Eventful entity) {
                 event.setTargetName(((Named) entity).getName());
             }
 
-            // TODO: FIXME:
-            // This is entirely useless. Our owner APIs expect the owner key, not the ID, so
-            // providing the ID does very little for event consumers.
             if (entity instanceof Owned) {
-                String ownerId = ((Owned) entity).getOwnerId();
-
-                if (ownerId != null) {
-                    event.setOwnerId(ownerId);
+                String ownerKey = ((Owned) entity).getOwnerKey();
```

Comment:
You can use pattern matching instanceOf to avoid manual casts.

Reply:
Good catch! Updated